### PR TITLE
Add reuseable workflows

### PR DESCRIPTION
**Pull request by RasmusSkytte**
This PR takes the [SCDB](https://github.com/ssi-dk/SCDB) github workflows and adds them to this repo.

Most of these workflows[^1] are converted to reusable workflows that other repositories can reference.
The process for each of these workflows involved splitting the workflows into a reuseable part (stored in `.github/workflows/`) and a referencing part (stored in `workflows/`). These referencing parts are the workflows the user should copy to the `.github/workflows/` directory of their own repository.

During this process, the names and trigger conditions for the workflows were adjusted slightly.

Furthermore, a single "super" workflow is added which calls all of the individual workflows.
The user can choose to copy this singular workflow to their repository. Then when we add other workflows in the future, this will be automatically added to their workflow stack.

I have added documentation on the workflows in the form of a README.md which should make their use more clear.

[^1]: `render-svg` and `license` workflows are not made reuseable

### CODE REVIEW

File: .github/workflows/code-coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,188 @@
+on:
+  workflow_call:
+    inputs:
+      run:
+        type: boolean
+        default: true
+      schemas:
+        type: string
+        default: ''
+      backend_skip:
```
Comment (marcusmunch) [inline .github/workflows/code-coverage.yaml:10]: GitHub Actions use `*_exclude` (`paths_exclude`, `branch_exclude`). For consistency, I would suggest changing `backend_skip` to align with this
```suggestion
      backend_exclude:
```

File: .github/workflows/code-coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage-test-failures
+          path: ${{ runner.temp }}/package
+
+
+  code-coverage-postgres:
+    name: 🧪 Tests (postgres)
+    runs-on: ubuntu-latest
+    if: ${{ inputs.run && !contains(inputs.backend_skip, 'postgres')}}
```
Comment (marcusmunch) [inline .github/workflows/code-coverage.yaml:59]:
```suggestion
    if: ${{ inputs.run && !contains(inputs.backend_exclude, 'postgres')}}
```

File: .github/workflows/code-coverage.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        with:
+          name: coverage-test-failures
+          path: ${{ runner.temp }}/package
+
+  # Option 1
+  # https://github.com/marketplace/actions/mssql-suite
+  code-coverage-mssql:
+    name: 🧪 Tests (mssql)
+    runs-on: ubuntu-latest
+    if: ${{ inputs.run && !contains(inputs.backend_skip, 'mssql')}}
```
Comment (marcusmunch) [inline .github/workflows/code-coverage.yaml:131]:
```suggestion
    if: ${{ inputs.run && !contains(inputs.backend_exclude, 'mssql')}}
```

File: workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+Determine the degree of [code coverage](https://en.wikipedia.org/wiki/Code_coverage) in your code base.
+
+#### Outcome
+Code coverage results are uploaded to [codecov.io](https://about.codecov.io/).
+This enables you to add the code coverage badge to your `README.md`.
+
+| Arguments      | Description                           | Default     | Example       |
+|----------------|---------------------------------------|-------------|---------------|
+| `schemas`      | Schemas to create (optional)          | ''          | test,test.one |
+| `backend_skip` | data base backends to skip (optional) | ''          | postgres      |
```
Comment (marcusmunch) [inline workflows/README.md:59]:
```suggestion
| `backend_exclude` | data base backends to skip (optional) | ''          | postgres      |
```

File: workflows/README.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+
+> [!NOTE]
+> To conserve resources, only the `code-coverage` workflow is configured with data bases.
+> This means that your `R-CMD-check` may not fail when `code-coverage` fails.
+> If this is the case, the cause is likely a data base specific issue in your code base.
```
Comment (marcusmunch) [inline workflows/README.md:65]: Note (most likely) to self: Is there a more elegant way to go about this?

* Trigger `R-CMD-check` (if defined for the repository) on `code-coverage` runs?
* Test if the current commit has triggered a `R-CMD-check` run, and trigger one if not?

File: workflows/README.md
Comment (marcusmunch) [inline workflows/README.md:1]: I would prefer "Exit status" over "Outcome", as it better describes the "content" of the section: When does the check pass (exit code 0) and when does it fail (non-zero status)?

File: workflows/render-svg.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        with:
+          packages: |
+            any::magick
+            any::rsvg
+
+      - name: Render logo SVG
+        run: |
+          library(magick)
+
+          svg_path <- list.files("man/figures", "*logo.svg", full.names = TRUE)
```
Comment (marcusmunch) [inline workflows/render-svg.yaml:28]: The `render-svg` workflow from ssi-dk/SCDB has recently been updated (https://github.com/ssi-dk/SCDB/commit/b7e5a4edd78a220c1435bc482f73744f8b43980f) to specifically look for `[package name]-logo.svg`.

```suggestion
          svg_path <- list.files(
            "man/figures",
            paste0(read.dcf("DESCRIPTION")[1, "Package"], "-logos.svg"),
            full.names = TRUE)
```

I would strongly recommend making these changes to this repository as well.

**Review by marcusmunch — APPROVED** [review]

File: workflows/render-svg.yaml (hunk 6 above)
Comment (RasmusSkytte) [reply workflows/render-svg.yaml:28]:
```suggestion
          svg_path <- list.files(
            "man/figures",
            paste0(read.dcf("DESCRIPTION")[1, "Package"], "-logo.svg"),
            full.names = TRUE
          )
```

**Pull request merged**